sdn_tests/testplans/Stress_Latency_Test_Plan.md

```diff
@@ -0,0 +1,385 @@
+
+## SET DELETE operation
+
+### Expectation
+
+The expectation is for the gNMI server to process the valid delete request and remove those nodes from the system. The server should send a valid response to the client once the request has been processed successfully.
+
+## GET operation
+
+### Expectation
+
+The GET response should contain the value of the request nodes. The expectation is for the gNMI server to process the valid get request and should send a valid response containing the values of the request nodes. Only the PROTO and JSON encodings are currently supported in the GPINS.
+
+## Subscribe operation
+
+### Expectation
+
+-   The system should be able to handle concurrent subscribe requests with different modes (ONCE/POLL/STREAM).
+-   The system should be able to reject invalid path( unsupported path )subscriptions. When the gNMI server implements the feature to silently accept the subscription for the unsupported path, the expectation for the unsupported path will change. This feature might be implemented in Q4 21.
+
+-   GET, SET(update, delete, replace), Subscription
+
+    -   Covered Subtrees
+
+<table>
+  <thead>
+    <tr>
+      <th><em>/</em></th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td><em>/interfaces</em></td>
+    </tr>
+    <tr>
+      <td><em>/components</em></td>
+    </tr>
+    <tr>
+      <td><em>/qos</em></td>
+    </tr>
+    <tr>
+      <td><em>/system</em></td>
+    </tr>
+    <tr>
+      <td><em>/lacp</em></td>
+    </tr>
+  </tbody>
+</table>
+
+The following test cases randomly select the set of paths from the above supported subtrees, the tests also randomly select the gNMI operations to validate the robustness of the system.\
+The key values are also randomly selected based on the testbed information. For example, the interface key for the interface module paths are randomly selected from the list of available interfaces in the testbed. The list of paths are randomly selected from the supported paths in go/gpins-openconfig.
+
+The infrastructure provides a fuzzing API to randomly select the list of paths, keys and gNMI operations to be validated along with the expected responses. The set of paths, keys, payload and expected response messages are provided as the artifact to the test cases.
+
+The list of paths will also include the paths which are not supported by the UMF and invalid keys. These paths are used in the test cases to validate the negative testing and gNMI server's performance metrics.
+
+The first few runs of these stress tests are used to establish the baseline benchmarks for each gNMI operation and for each test case. The baseline benchmarks are stored in the table along with the list of gNMI paths, payloads, keys and expected response. There will be another test which runs at the end of each test which analyzes the performance report against the baseline benchmarks for that system.The test will fail if the performance exceeds the benchmarks.
```

Comment:
The sentences on line 99 can be removed. I do not believe our Ondatra tests do this today.